json_options.go

```diff
@@ -0,0 +1,88 @@
+package zap
+
+import "time"
+
+// JSONOption is used to set options for a JSON encoder.
+type JSONOption interface {
```

Comment:
I'd also unexport the `Apply` method so that the custom formatter types won't display `Apply` in the documentation